### PR TITLE
Add info about JIRA links to example RST file in style guide

### PR DESCRIPTION
**Pull request by srpearce**
## [DOC-2990](https://openedx.atlassian.net/browse/DOC-2990)

This PR adds information about creating JIRA links to the ExampleRSTFile.rst file in the style guide. It also clarifies how to create internal and external references and removes outdated information about creating links to edX101.
- [ ] doc team review (copy edit and sanity check): @lamagnifica or @pdesjardins

### CODE REVIEW

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 1]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
+that is titled "Keyboard Shortcuts for Notes", and "keyboard shortcuts" is the
+link text. Users will see the following text, and "keyboard shortcuts" will be
+an active link.
 
 ::
 
-  For more information, see :ref:`the introductory section on
-  exercises<Exercises_introduction>`
+  If you want to, you can use keyboard shortcuts to create, edit, and view your
+  notes.
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst:394]: I'd suggest another topic here on intersphinx mapping: how to find the map IDs (in edx-documentation/shared/conf.py) and how to create inter-guide references by adding the ID:anchor for the destination.

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 2]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
+external websites.
+
+==========================
+Internal Cross-References
+==========================
+
+Cross-references for topics in the edX documentation, such as the *Building and
+Running an edX Course* guide and the XBlock tutorial, use anchors that are
+placed above the heading for the target section. Anchors are defined in lines
+beginning with 2 periods, followed by a space, underscore and the anchor text,
```
Comment (pdesjardins) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: I would spell out two based on previous style guide reading, but I'm sure you didn't for a reason. MMOS seems to be flexible. So, no real suggestion here.

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 3]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
+and ending with a colon.
 
 For example, ``.. _Anchors and Internal Links:``
 
-You can also see examples of anchors above the first three headings in this document.
+You can also see examples of anchors above the first three headings in this
+document.
 
 Anchor text is never visible in output. It is always replaced either by the
 text of the anchored topic heading, or by the specified link text.
```
Comment (pdesjardins) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst:336]: Maybe add an external reference to the Sphinx documentation of the :ref: directive: http://www.sphinx-doc.org/en/stable/markup/inline.html#cross-referencing-arbitrary-locations

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 4]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
-   Ensure that your course introduction video follows the same
-   :ref:`Compression Specifications` and :ref:`Video Formats` guidelines as
-   course content videos
+   For more information about using keyboard shortcuts, see :ref:`SFD SN
+   Keyboard Shortcuts`.
+
+In this example, "SFD SN Keyboard Shortcuts" is the anchor text for a section
+that is titled "Keyboard Shortcuts for Notes". In output, the actual text of
+the associated headings is substituted, so that users see the following text.
+"Keyboard Shortcuts for Notes" will be an active link.
```
Comment (pdesjardins) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: > the following text.

I paused here, trying to figure out how the next sentence was formed from the example source. Which makes no sense, but it held me up. Move '"Keyboard Shortcuts for Notes" will be an active link' below the example so the example follows "following text"?

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 5]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
+  If you want to, you can use :ref:`keyboard shortcuts<SFD SN Keyboard
+  Shortcuts>` to create, edit, and view your notes.
 
-=======================================================
-Example of Cross-Reference Using Specified Link Text
-=======================================================
+In this example, "SFD SN Keyboard Shortcuts" is the anchor text for a section
+that is titled "Keyboard Shortcuts for Notes", and "keyboard shortcuts" is the
+link text. Users will see the following text, and "keyboard shortcuts" will be
+an active link.
```
Comment (pdesjardins) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: Oh, this structure works better for me. Use it in the section above?

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 6]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
+In edX documentation, you can include cross-references to other topics in the
+edX guides (also called internal cross-references), to JIRA stories, and to
+external websites.
+
+==========================
+Internal Cross-References
+==========================
+
+Cross-references for topics in the edX documentation, such as the *Building and
+Running an edX Course* guide and the XBlock tutorial, use anchors that are
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: these two guides, in conjunction, and with different formatting, strike me as just odd. Maybe stick with one example?

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 7]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
+placed above the heading for the target section. Anchors are defined in lines
+beginning with 2 periods, followed by a space, underscore and the anchor text,
+and ending with a colon.
 
 For example, ``.. _Anchors and Internal Links:``
 
-You can also see examples of anchors above the first three headings in this document.
+You can also see examples of anchors above the first three headings in this
+document.
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: ... if you look in the source rst file that is!

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 8]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
 
 Anchor text is never visible in output. It is always replaced either by the
 text of the anchored topic heading, or by the specified link text.
 
-=================================================
-Example of Cross-Reference Using Anchor Only
-=================================================
+
+Internal Cross-Reference Using Anchor Only
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: Creating an...
Formatting for an...
make active somehow?

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 9]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
 
 ============================================
 Cross-References to External Web Pages
 ============================================
 
 To reference an external web page, you decide what text to use for the cross-
-reference, such as the title of the web page, and add that text using markup.
-Then, you add that same text to the links.rst file along with the actual URL.
-An example follows.
+reference, and add that text using markup. Then, you add that same text to the
```
Comment (pdesjardins) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: that same text > the same text

My personal preference, I think.

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 10]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
 text of the anchored topic heading, or by the specified link text.
 
-=================================================
-Example of Cross-Reference Using Anchor Only
-=================================================
+
+Internal Cross-Reference Using Anchor Only
+********************************************
 
 For cross-references that use the actual text of the target topic’s heading,
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: use --> present?
maybe rephrase somehow to make it clearer that the end result is for the rendered text to show the full heading that follows the anchor, as distinct from the next kind, where you replace the header with your own text...

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 11]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
 ============================================
 Cross-References to External Web Pages
 ============================================
 
 To reference an external web page, you decide what text to use for the cross-
-reference, such as the title of the web page, and add that text using markup.
-Then, you add that same text to the links.rst file along with the actual URL.
-An example follows.
+reference, and add that text using markup. Then, you add that same text to the
+links.rst file along with the actual URL. An example follows.
```
Comment (pdesjardins) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: Give more information about where the links.rst file is, in case someone doesn't know. Maybe "to the /en_us/links/links.rst file"?

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 12]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
 ::
 
-   Ensure that your course introduction video follows the same
-   :ref:`Compression Specifications` and :ref:`Video Formats` guidelines as
-   course content videos
+   For more information about using keyboard shortcuts, see :ref:`SFD SN
+   Keyboard Shortcuts`.
+
+In this example, "SFD SN Keyboard Shortcuts" is the anchor text for a section
+that is titled "Keyboard Shortcuts for Notes". In output, the actual text of
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: In ^the output 
or
In ^the rendered output

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 13]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
+the associated headings is substituted, so that users see the following text.
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: users --> readers

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 14]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
+"Keyboard Shortcuts for Notes" will be an active link.
+
+::
+
+  For more information about using keyboard shortcuts, see Keyboard Shortcuts
+  for Notes.
+
+
+Internal Cross-Reference Using Specified Link Text
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: Specifying the Linked Text in an Internal Cross-reference (?)

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 15]
```diff
@@ -308,55 +308,93 @@ Notes and Warnings
...
+****************************************************
+
+For internal cross-references that use text other than the heading for the
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst:372]: An option for internal cross-references is to specify the text of the link, instead of using the heading... To do so, use...

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 16]
```diff
@@ -391,19 +429,21 @@ An example follows.
 
       .. _Release Pages: https://openedx.atlassian.net/wiki/display/ENG/Release+Pages
 
-============================================
-Cross-References to edX101 Course Pages
-============================================
+Users will see the following text. "Release Pages" in the first sentence will
```
Comment (pdesjardins) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: I like the structure that combines these sentences.

File: en_us/edx_style_guide/source/ExampleRSTFile.rst (hunk 10 above)
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: also, maybe explicitly state that this is true for intra-guide xrefs only, not across guides?

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 17]
```diff
@@ -391,19 +429,21 @@ An example follows.
...
+      The edX engineering wiki Release Pages provide access to detailed
+      information about every change made to the edx-platform GitHub
+      repository. If you are interested in additional information about every
+      change in a release, create a user account for the wiki and review the
+      dated release pages.
+
+Note that the text that you specify in markup and in the links.rst file will be
+the verbatim text in the document, including capitalization. If you want to use
+the title of a web page as the link text, make sure to capitalize the link text
+in markup and the links.rst file the way you want it to appear in the output.
```
Comment (pdesjardins) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: We could include a reference to the RST spec for these external references: http://docutils.sourceforge.net/docs/user/rst/quickref.html#hyperlink-targets.

Also, if your .rst file doesn't include the include directive for the links/links.rst file, you'll need to add it (and figure out the relative file path from your .rst file to links.rst).

File: en_us/edx_style_guide/source/ExampleRSTFile.rst (hunk 8 above)
Comment (pdesjardins) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: Pluralize "Cross-Reference" to match other headings.

**Comment by pdesjardins** [comment]
Looks good, thanks! I added a few comments. 👍

File: en_us/edx_style_guide/source/ExampleRSTFile.rst (hunk 1 above)
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst:394]: outside the scope of this change, perhaps...

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 18]
```diff
@@ -391,19 +429,21 @@ An example follows.
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst:519]: on line 412, These links are typically ---> These `include` directives are typically

File: en_us/edx_style_guide/source/ExampleRSTFile.rst (hunk 11 above)
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: perhaps mention in this paragraph that for accessibility/readability, a link to a web page title is preferred over a link to a work like "here"

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 19]
```diff
@@ -391,19 +429,21 @@ An example follows.
...
-   /courses/edX/edX101/How_to_Create_an_edX_Course/courseware/a45de3baa8a9468c
-   bfb1a301fdcd7e86/3ba055e760d04f389150a75edfecb844/1>`_
+      The edX engineering wiki Release Pages provide access to detailed
+      information about every change made to the edx-platform GitHub
+      repository. If you are interested in additional information about every
+      change in a release, create a user account for the wiki and review the
+      dated release pages.
+
+Note that the text that you specify in markup and in the links.rst file will be
+the verbatim text in the document, including capitalization. If you want to use
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: "the verbatim text in the document" is confusing. Do you mean in the output, in the web page you're referencing, in the link text?
I think the first sentence could just focus on the need for the link text that you put in the links.rst file to match the link text that you mark up in another rst file **exactly**.

**Comment by lamagnifica** [comment]
a bunch of suggestions

**Comment by srpearce** [comment]
@lamagnifica @catong @pdesjardins Made some updates - please review again.

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 20]
```diff
@@ -308,71 +308,188 @@ Notes and Warnings
...
 
-Cross-references use anchors that are placed above the heading for the target
-section. Anchors are defined in lines beginning with 2 periods, followed by a
-space, underscore and the anchor text, and ending with a colon.
+In edX documentation, you can include cross-references to other locations in
+the same edX document, to locations in other edX documents (such as a cross-
+reference from a location in the *Building and Running an edX Course* guide to
+a location in the *EdX Learner's Guide*), to JIRA stories, and to external
+websites. "EdX documents" refers to the resources, including guides and
+tutorials, listed on docs.edx.org.
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: locations --> sections and topics (or guides?)
(locations seems really odd to me, like we're talking about the Bahamas or Chattanooga or something...!)
Perhaps introduce the last "edX documents" sentence with something like, "In this section,": In this section, "edX documents" refers to the resources, including guides and tutorials, ^that are listed on docs.edx.org.
Perhaps use In edX documents in the first sentence instead of edX documentation?

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 21]
```diff
@@ -308,72 +308,107 @@ Notes and Warnings
...
+Cross-References to Locations in Different Documents
+====================================================
 
-To create an internal cross-reference from one document to another, you use
-intersphinx mapping to specify the map ID of the document that you want to link
-to, and then specify the anchor text of the topic you want. The cross-reference
-uses the following syntax.
+You can create cross-references between different edX documents. "EdX
+documents" refers to the resources, including guides and tutorials, listed on
+docs.edx.org. For example, you can create a link in *Building and Running an
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: maybe no need to repeat this 2nd sentence?

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 22]
```diff
@@ -308,72 +308,107 @@ Notes and Warnings
...
+reference from a location in the *Building and Running an edX Course* guide to
+a location in the *EdX Learner's Guide*), to JIRA stories, and to external
+websites. "EdX documents" refers to the resources, including guides and
+tutorials, listed on docs.edx.org.
 
-==========================
-Internal Cross-References
-==========================
+==================================================
+Cross-References to Locations in the Same Document
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst:320]: add verb or make Cross-Reference the verb instead of a noun

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 23]
```diff
@@ -308,72 +308,107 @@ Notes and Warnings
...
 
 Anchor text is never visible in output. It is always replaced either by the
-text of the anchored topic heading, or by the specified link text.
+text of the anchored topic heading, or by link text that you specify.
 
 For more information, see `Cross-referencing arbitrary locations`_ in the
 online Sphinx documentation.
 
-Internal Cross-References Using Anchor Only
+Cross-References Using the Anchor Only
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst:348]: add verb

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 24]
```diff
@@ -308,72 +308,107 @@ Notes and Warnings
...
+text, and "Keyboard Shortcuts for Notes" will be an active link.
 
 ::
 
   For more information about using keyboard shortcuts, see Keyboard Shortcuts
   for Notes.
 
-Internal Cross-References to Other EdX Guides
-==============================================
+Cross-References Using Specified Link Text
```
Comment (lamagnifica) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst:369]: verb (throughout)

**Comment by lamagnifica** [comment]
thanks for bringing clarity to the different flavors of xref... a few more suggestions, ship when ready!

File: en_us/edx_style_guide/source/ExampleRSTFile.rst (hunk 22 above)
Comment (srpearce) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst:320]: The other headings and subheadings in this topic use nouns instead of verbs (e.g., "Nested Lists or Content," "Tables", "Code, Images, and Other Content inside Lists") - I'd prefer to keep this consistent.

File: en_us/edx_style_guide/source/ExampleRSTFile.rst (hunk 20 above)
Comment (srpearce) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: Heh - I hadn't thought about Chattanooga. :-) I'd prefer to keep "location" just because it's shorter than "topic or section" - replacing each instance of "location" with "topic or section" could be pretty unwieldy.

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 25]
```diff
@@ -308,71 +308,188 @@ Notes and Warnings
...
+============================================
+
+For cross-references to JIRA tickets, use the ``:jira:`PROJECT-ticketnumber```
+syntax, as in the following example.
+
+::
+
+  :jira:`DOC-2316`
+
+Note that ``jira`` replaces ``ref`` for these links.
```
Comment (catong) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst:466]: @srpearce Can we add a bit more information about how the output appears, as you do for the cross-document section, e.g. "DOC-2316" will be an active link to the JIRA story in Confluence".

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 26]
```diff
@@ -308,71 +308,188 @@ Notes and Warnings
...
 
-#. In the paragraph where you want the cross-reference, create an anchor that
-   represents the text that you want to use for the cross-reference.
+* The URL of the external web page.
+* The text to use for the cross-reference. This text becomes an anchor in the
+  file that contains the cross-reference.
+* An ``include`` directive in the file that contains the cross-reference to the
+  links.rst file that is located in the ``edx-documentation/en_us/links/``
+  folder.
+* An enry in the links.rst file.
```
Comment (catong) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst]: Typo "entry"

File: en_us/edx_style_guide/source/ExampleRSTFile.rst [hunk 27]
```diff
@@ -308,71 +308,188 @@ Notes and Warnings
...
+The following example shows an anchor for a section, followed by the heading
+for that section.
+
+::
+
+  .. _SFD SN Keyboard Shortcuts:
+
+  ******************************
+  Keyboard Shortcuts for Notes
+  ******************************
```
Comment (catong) [inline en_us/edx_style_guide/source/ExampleRSTFile.rst:341]: Add a warning against using punctuation in anchors?

**Comment by catong** [comment]
Thanks for doing this @srpearce I made a couple of suggestions in comments, but 👍
